File: src/dependent/payments/TaxCalculator.java
package org.example.dependent.payments;

import java.math.BigDecimal;

public class TaxCalculator {
    public BigDecimal calculateTotalWithTax(Cart cart, String currency) {
        if (cart == null) {
            throw new IllegalArgumentException("Cart cannot be null");
        }
        BigDecimal subtotal = cart.getSubtotal();
        BigDecimal taxRate = new BigDecimal("0.20");
        BigDecimal tax = subtotal.multiply(taxRate);
        return subtotal.add(tax);
    }
}

File: src/dependent/payments/tax_calculator.py
from decimal import Decimal

from .cart import Cart


class TaxCalculator:
    def calculate_total_with_tax(self, cart: Cart, currency: str) -> Decimal:
        if cart is None:
            raise ValueError("Cart cannot be null")
        subtotal = cart.subtotal
        tax_rate = Decimal("0.20")
        tax = subtotal * tax_rate
        return subtotal + tax
